packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/test/Providers/NamedTypeSymbolProviders/ConstructorInitializerTests.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Tests.Common;
+using NUnit.Framework;
+
+namespace Microsoft.TypeSpec.Generator.Tests.Providers.NamedTypeSymbolProviders
+{
+    public class ConstructorInitializerTests
+    {
+        [SetUp]
+        public void Setup()
+        {
+            MockHelpers.LoadMockGenerator();
+        }
+
+        [Test]
+        public async Task CustomConstructorWithThisInitializerShouldHaveInitializerPopulated()
+        {
+            // Arrange
+            var compilation = await Helpers.GetCompilationFromDirectoryAsync();
```

Comment:
We should be passing the compilation arg to the LoadMockGenerator call. Let's remove the separate Setup method, and call LoadMockGeneratorAsync in each test. We can then create a ModelProvider instance that has the same name as the customization class. Similar to here - https://github.com/microsoft/typespec/blob/main/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/test/Providers/ModelProviders/ModelCustomizationTests.cs#L29-L30